### DISCUSSION TITLE
Flash PIC firmware by USB and software microchip-need help

### DISCUSSION THREAD

**Discussion by pepsonEL** [john30/ebusd#566] [General]
Hi
I try flash my adapter by Bootloader Host Application from Microchip... I setup J11 pin3 and pin 4 by jumper and try. But no succesfully. This is screen,. Setup serial USB to speed 115200
now my adapter no working.... Please help me
![156412258-315cad90-432b-442d-b79a-619ea0f2b665](https://user-images.githubusercontent.com/71824185/156437195-22d32127-f928-42e7-833a-88642228d3cf.jpg)
.

**Comment by john30** [comment]
use ebuspicloader and not the crappy microchip solution

**Reply by pepsonEL** [reply]
But how use it because I don't have raspberry. I have only notebook with windows.

**Comment by john30** [comment]
where does ebusd run then? on the same host you can use ebuspicloader as well

**Reply by pepsonEL** [reply]
I run ebus adapter with wemos wifi and ebusd run as addon on Home Assistant or i can run it in docker on Synology NAS.

**Comment by pepsonEL** [comment]
But maybe on software microchip only change address and size ? Maybe it is wrong but i dont know what?
Bootloader offset(Byte address) : 0x600
EEPROM offset(byte address): 0x1E000
Config offset(byte address): 0x10000

Program memory size(byte): 0x4000
EEPROM memory size(bytes):0x40
Config memory size(bytes): 0x0E

It is default when run software for PIC16
